### DISCUSSION TITLE
IPerf3

### DISCUSSION THREAD

**Discussion by mamer11** [esnet/iperf#1230] [Q&A]
Hello,

Is there any command to run IPerf3 continuously (2-3 days) in Linux, I tried to use -t in command, but seems if the duration is big then the command is not working.

Thanks

**Comment by davidBar-On** [comment]
If I understand correctly, test time value is limited to one day (86400 sec), but setting the test time to zero by `-t 0` means unlimited test time (test has to be terminated by stopping the client manually).

**Comment by mamer11** [comment]
Thank you. I applied both, I'm running now 86400 sec, it's working, but -t 0 worked for a few seconds then stopped automatically.

**Comment by davidBar-On** [comment]
> -t 0 worked for a few seconds then stopped automatically.

This is strange.  Before sending the response I run the client with `-t 0` for over 5 minutes (and terminated it manually) just to make sure it works.  Did the client or server issue any error message?  If not, can you try re-running with `-v -t 0` and see if there is any error message?

**Comment by mamer11** [comment]
when I applied -t 0 I didn't get an error message but in -v -t 0 I got the below error message: 
iperf3: error - control socket has closed unexpectedly

**Comment by davidBar-On** [comment]
Sorry, I see that I wrote `-v` instead of `-V` (Capital V is verbose mode).  Can you try running both server and client with `-V` or even `-V -d` to get debug information?

Did the test star send data packets before the error occurred?  Also, did you see any error on the server side?  Usually that error happens when the server is terminating the test for some reason.

In addition, which OS you are using and what iperf3 version (`iperf3 -V`)?

**Comment by swlars** [comment]
You might want to try `iperf3 -c -t inf` for longer durations.
